### PR TITLE
[PM-212] Sync Organization Billing Email from Stripe Webhook

### PR DESCRIPTION
**Pull request by amorask-bitwarden**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

[PM-212](https://bitwarden.atlassian.net/browse/PM-212)

This PR has two purposes:
1. Refactor out various private methods within the `StripeController` into a `StripeEventService` for better maintainability and testability.
2. Begin handling the Stripe `customer.updated` event in the `StripeController` to update an organization's billing email when an organization customer is updated in Stripe.

## Code changes
Since some of the code changes are self explanatory, I'll go over the important ones here.

### src
* **StripeFacade.cs:** A mockable facade in the Billing project used to aggregate various Stripe service methods.
* **StripeEventService.cs:**: A service used to retrieve or check various parts of an incoming Stripe `Event` object.
* **StripeController.cs:**: Replaced various private method invocations operating on Stripe `Event` objects with calls to the `StripeEventService`. Added handling for the `customer.updated` event to check if the customer updated is an organization and, if so, update the organization's billing address with the email in the incoming webhook.

### test
* **Resources/Events:** A folder used to hold embedded JSON resources representing Stripe webhook events taken from the Test Stripe environment.
* **EmbeddedResourceReader.cs:** A static class used to read the resources in the folder mentioned above.
* **StripeTestEvents:** A static class used that uses the `EmbeddedResourceReader` to read the `Event` JSON and turn them into testable Stripe `Event` objects using `EventUtility.ParseEvent`. 

https://github.com/bitwarden/server/assets/144709477/88f5e89c-3113-46e8-b9e0-f8b64409bd25

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

[PM-212]: https://bitwarden.atlassian.net/browse/PM-212?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by djsmith85** [comment]
I assume this PR and https://github.com/bitwarden/server/pull/3319 will largely require reviews from @bitwarden/team-billing-dev and not just from @bitwarden/team-tools-dev.

@cturnbull-bitwarden Could you please look into adding CODEOWNERS for the billing-owned components.

**Review by cturnbull-bitwarden — APPROVED** [review]
👍 Nice work!

File: src/Core/Tools/Enums/ReferenceEventType.cs
Comment (djsmith85) [inline src/Core/Tools/Enums/ReferenceEventType.cs:46]: I don't see this used anywhere? Is this part of some other work?

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: src/Core/Tools/Enums/ReferenceEventType.cs
Comment (djsmith85) [reply src/Core/Tools/Enums/ReferenceEventType.cs:46]: Dicussed this with @amorask-bitwarden and I just missed the usage in the StripeController. Changes are looking good.

**Review by djsmith85 — APPROVED** [review]
Approving changes for @bitwarden/team-tools-dev

**Pull request merged**
